app/client/components/buildViewSectionDom.ts

```diff
@@ -129,7 +130,12 @@ export function buildViewSectionDom(options: {
           dom('div.viewsection_truncated', t('Not all data is shown'))
         ),
         dom.cls((use) => 'viewsection_type_' + use(vs.parentKey)),
-        viewInstance.viewPane
+        viewInstance.viewPane,
+        dom.domComputed(use =>
+          (use(viewInstance.viewSection.hasFocus) && use(viewInstance.enableAddRow)),
+          (showNewRecordButton) =>
+            showNewRecordButton ? newRecordButton(viewInstance) : null
+        )
```

Comment:
This is out of the `viewPane` mostly because some views have `overflow: hidden` on their wrapper and since we want the button to be a bit out of the view it needs to be outside the wrapper.